dialogue-target/src/main/java/com/palantir/dialogue/UrlBuilder.java

```diff
@@ -59,35 +61,50 @@ private UrlBuilder(String protocol) {
      * {@code [2010:836B:4179::836B:4179]} (note the enclosing square brackets).
      */
     public UrlBuilder host(String theHost) {
+        Preconditions.checkArgument(UrlEncoder.isHost(theHost), "invalid host format", UnsafeArg.of("host", theHost));
         this.host = theHost;
         return this;
     }
 
     public UrlBuilder port(int thePort) {
+        Preconditions.checkArgument(thePort >= 0 && thePort <= 65535, "port must be in range [0, 65535]");
         this.port = thePort;
         return this;
     }
 
+    /**
+     * Adds the given URL-encoded path segment (one or more) to the list of segments, or fails if the given segments
+     * contain forbidden characters. Note that leading or trailing slashes are preserved, for instance {@code url
+     * .pathSegment("foo").encodedPathSegments{"/bar/"}.pathSegment("baz")} yields a URL with path
+     * {@code foo//bar//baz} (note the empty segments).
+     */
+    public UrlBuilder encodedPathSegments(String segments) {
```

Comment:
the method naming here is a little confusing. I think this "encoded" in the sense that the argument is already encoded? Not sure I have a better suggestion, but seems like it's going to catch someone offguard down the road.

Reply:
yeah, I agree, but I also don't have a better proposal. maybe `preEncodedPathSegments`? I added a fair amount of docs and an example above, at least.